src/Adapter/Session/Repository/CustomerSessionRepository.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Session\Repository;
+
+use CustomerSession;
+use DateInterval;
+use DateTime;
+use Doctrine\DBAL\Connection;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelRepository;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotBulkDeleteCustomerSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotClearCustomerSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotDeleteCustomerSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\SessionNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Security\ValueObject\CustomerSessionId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+/**
+ * Methods to access data storage for Customer session
+ */
+class CustomerSessionRepository extends AbstractObjectModelRepository
```

Comment:
I should be missing something but I can't find where this is constructed/declared as a service. Also, what's the unit for `$cookieLifetime`? From what I see in `clearOutdatedSessions()` it seems to be in Hour but maybe a comment to mention that would be nice :)

Reply:
`prestashop.adapter.security.repository.customer_session_repository` is the service name.
The cookie lifetime is the same as the one defined in the Core, so yes, it's in hour. I can add a comment if you think it's really needed.